pkg/controller/installation/core_controller.go

```diff
@@ -1783,3 +1779,35 @@ func removeInstallationFinalizer(i *operator.Installation) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(CalicoFinalizer, i.GetFinalizers()))
 	}
 }
+
+func (r *ReconcileInstallation) updateInstallationDegraded(reason string, err error, instance *operator.Installation, ctx context.Context, log logr.Logger) {
```

Comment:
Is there a reason you create a function, rather than expanding the existing r.SetDegraded(...)? I believe it would be better to move this code to the other method.